go-controller/pkg/libovsdb/libovsdb.go

```diff
@@ -31,7 +36,40 @@ import (
 func newClient(cfg config.OvnAuthConfig, dbModel model.ClientDBModel, stopCh <-chan struct{}, opts ...client.Option) (client.Client, error) {
 	const connectTimeout time.Duration = types.OVSDBTimeout * 2
 	const inactivityTimeout time.Duration = types.OVSDBTimeout * 18
-	logger := klogr.New()
+
+	var logger logr.Logger
+
+	logggerFilename := config.Logging.LibovsdbFile
+	if len(logggerFilename) == 0 {
+		// Not using a separate log file for libovsdb client
+		logger = klogr.New()
+	} else {
+		// Sanity check: Make sure logger file can be opened
+		err := os.MkdirAll(filepath.Dir(logggerFilename), 0744)
+		if err != nil {
+			return nil, fmt.Errorf("making directories for logger file %s for libovsdb failed: %w", logggerFilename, err)
+		}
+		checkFile, err := os.OpenFile(logggerFilename, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			return nil, fmt.Errorf("opening logger file %s for libovsdb failed: %w", logggerFilename, err)
+		}
+		_ = checkFile.Close()
```

Comment:
lumberjack is already creating the dir/file for you. It is using 0644 for files by default.
I wouldn't do it here unless we need more restrictive permissions and I am thinking that perhaps we do, like 0640

Reply:
Good point. I was being a bit paranoid about it. Let's remove this.
Less places for a bug to live. Less is more.

Reply:
But what do you think though about what permissions should we use?

Reply:
oh; did not think about that. I'd think we should use the same perms we have for the ovnkube-master logs before we had this change. I will see if that is configurable and compare against what we currently use.

Reply:
interesting... this is not configurable:  https://github.com/natefinch/lumberjack/blob/4cb27fcfbb0f35cb48c542c5ea80b7c1d18933d0/lumberjack.go#L215C14-L215C14
And it seems that it has changes on recent version of lumberchack:
https://github.com/natefinch/lumberjack/commit/2e8fbeea3f9e456ee5062dfbcbe3ee902a44bace

Reply:
https://github.com/natefinch/lumberjack/issues/82#issuecomment-482143273

Reply:
@jcaamano  I re-instated the changes for opening the file with the perms that we want. PTAL